include/seqan/index/find2_index_approx.h

```diff
@@ -295,16 +331,89 @@ inline void _optimalSearchSchemeComputeFixedBlocklength(std::array<OptimalSearch
+                               uint8_t const blockIndex,
+                               TDir const & /**/)
+{
+    auto const & genome = indexText(*iter.fwdIter.index);
+
+    //cut of blockStarts and Ends that where already checked by the search
+    std::vector<uint32_t> blockStarts(s.pi.size() - blockIndex);
+    std::vector<uint32_t> blockEnds(s.pi.size() - blockIndex);
+    std::copy(std::begin(s.blockStarts) + blockIndex, std::end(s.blockStarts), std::begin(blockStarts));
+    std::copy(std::begin(s.blockEnds) + blockIndex, std::end(s.blockEnds), std::begin(blockEnds));
+
+    // modify blockStart or blockEnd if we are already inside a block
+    if (std::is_same<TDir, Rev>::value)
+    {
+        if (needleRightPos - 1 > blockStarts[0] && needleRightPos - 1 < blockEnds[0])
+            blockStarts[0] = needleRightPos - 1;
+    }
+    else
+    {
+        if (needleLeftPos > blockStarts[0] && needleLeftPos < blockEnds[0])
+            blockEnds[0] = needleLeftPos;
+    }
+
+    // iterate over each potential occurrence inside the range
+    for (uint32_t i = iter.fwdIter.vDesc.range.i1; i < iter.fwdIter.vDesc.range.i2; ++i)
+    {
+        bool valid = true;
+        Pair<uint16_t, uint32_t> sa_info = iter.fwdIter.index->sa[i];
+        uint32_t chromlength = length(genome[sa_info.i1]);
+        if (!(needleLeftPos <= sa_info.i2 && chromlength - 1 >= sa_info.i2 - needleLeftPos + length(needle) - 1))
+            continue;
+
+        sa_info.i2 = sa_info.i2 - needleLeftPos;
+        uint8_t errors2 = errors;
+	// iterate over each block according to search in search scheme
+        for (uint32_t j = 0; j < blockStarts.size(); ++j)
+        {
+            // compare bases to needle
+            for (uint32_t k = blockStarts[j]; k <  blockEnds[j]; ++k)
+            {
+                if (needle[k] != genome[sa_info.i1][sa_info.i2 + k])
+                    ++errors2;
+            }
+            // check after each block if errors conform with the search in search scheme
+            if (errors2 < s.l[blockIndex + j] || errors2 > s.u[blockIndex + j])
+            {
+                valid = false;
+                break;
+            }
+        }
+        if (valid)
+            delegateDirect(sa_info, needle, errors2);
+    }
+}
+
+
+template <typename TDelegate, typename TDelegateD, typename TCondition,
           typename TText, typename TIndex, typename TIndexSpec,
           typename TNeedle,
           size_t nbrBlocks,
           typename TDir>
 inline void _optimalSearchSchemeDeletion(TDelegate & delegate,
-                                         Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                                         TDelegateD & delegateDirect,
+                                         TCondition & itvCondition,
+                                         Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<
+                                         TIndexSpec> > > iter,
```

Comment:
remove line break if it doesn't exceed 120 chars. If it does at least indent this.